examples/mecenas.cash

```diff
@@ -1,37 +1,29 @@
-pragma cashscript ^0.6.0;
+pragma cashscript ^0.7.0;
 
 /* This is an unofficial CashScript port of Licho's Mecenas contract. It is
  * not compatible with Licho's EC plugin, but rather meant as a demonstration
  * of covenants in CashScript.
  * The time checking has been removed so it can be tested without time requirements.
  */
-contract Mecenas(bytes20 recipient, bytes20 funder, int pledge/*, int period*/) {
-    function receive(pubkey pk, sig s) {
-        require(checkSig(s, pk));
-
+contract Mecenas(bytes20 recipient, bytes20 funder, int pledge/*, int period */) {
+    function receive() {
         // require(tx.age >= period);
 
-        int minerFee = 1000;
-        int intValue = int(bytes(tx.value));
+        // Check that the first output sends to the recipient
+        require(tx.outputs[0].lockingBytecode == new LockingBytecodeP2PKH(recipient));
 
-        if (intValue <= pledge + minerFee) {
-            /* The contract has less value than the pledge, or equal.
-             * The recipient must claim all of of it. */
+        int minerFee = 1000;
+        int currentValue = tx.inputs[this.activeInputIndex].value;
+        int changeValue = currentValue - pledge - minerFee;
 
-            bytes8 amount1 = bytes8(intValue - minerFee);
-            bytes34 out1 = new OutputP2PKH(amount1, recipient);
-            require(hash256(out1) == tx.hashOutputs);
+        // If there is not enough left for *another* pledge after this one, we send the remainder to the recipient
+        // Otherwise we send the remainder to the recipient and the change back to the contract
+        if (changeValue <= minerFee * 2) {
```

Comment:
```suggestion
        if (changeValue <= pledge + minerFee) {
```
Or did I misunderstand?

Reply:
Oh huh. Wow, yeah I'm pretty sure you're right there, good catch. 😅

Reply:
Fixed this.